master/internal/api_shell.go

```diff
@@ -106,7 +117,19 @@ func (a *apiServer) KillShell(
 func (a *apiServer) SetShellPriority(
 	ctx context.Context, req *apiv1.SetShellPriorityRequest,
 ) (resp *apiv1.SetShellPriorityResponse, err error) {
-	if _, err := a.GetShell(ctx, &apiv1.GetShellRequest{ShellId: req.ShellId}); err != nil {
+	getResponse, err := a.GetShell(ctx, &apiv1.GetShellRequest{ShellId: req.ShellId})
+	if err != nil {
+		return nil, err
+	}
+
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = command.AuthZProvider.Get().CanSetNSCsPriority(
+		ctx, *curUser, model.AccessScopeID(getResponse.Shell.WorkspaceId), int(req.Priority))
+	if err != nil {
```

Comment:
these can leak that the entity exists could you hook it up to tests here https://github.com/hamidzr/determined/blob/b6f7f5ea0005a9a2e39fa6a0a6c3c9fdb3f810fb/master/internal/api_ntsc_intg_test.go#L176 ? 

although looking at it again the assertion in the test probably has to be more complicated based on if the user has view access or not. but defaulting to 404 seems like a saner option?

Reply:
sg, I'll add tests and check out the response. I have yet to modify things into 404 errors.